### PR TITLE
Fixed problems with database request/response/gateway classes

### PR DESCRIPTION
**Pull request by Spaghetti-12**
Changed all of the classes in the db package from records to actual classes as requested by the TA. Also made some tune-ups and added comments to some of the classes I'd previously written in the BuyStockUseCase and APIInterface packages.

### CODE REVIEW

**Comment by eladperlman** [comment]
Add private to all variables in the request and response objects your changes, as well as in the gateway you used the public variables for portfolio request instead of its getter methods. Also, you should make an interface for the stock API so that when you instantiate it you aren't using the specific implementation (dependency inversion), look at my branch on how userManager interacts with the database gateway for reference. Finally, you shouldn't be passing string messages as your response, you should be passing in only the minimum needed information. For example what if we want to notify the user of the error through a vibration or something and not visual message?

**Comment by Spaghetti-12** [comment]
All done! Thanks for the input

File: src/main/java/BuyStockUseCase/BuyInputRequest.java [hunk 1]
```diff
@@ -8,7 +8,7 @@ public class BuyInputRequest {
      * Request model class, carries user input to be passed to the BuyUseCaseInteractor.
      */
 
-    String symbol;
+    private String symbol;
     int buy_quantity;
```
Comment (eladperlman) [inline src/main/java/BuyStockUseCase/BuyInputRequest.java:12]: should all be private

File: src/main/java/BuyStockUseCase/BuyInputRequest.java [hunk 2]
```diff
@@ -8,7 +8,7 @@ public class BuyInputRequest {
...
     Portfolio port;
```
Comment (eladperlman) [inline src/main/java/BuyStockUseCase/BuyInputRequest.java:13]: same

File: src/main/java/BuyStockUseCase/BuyInputRequest.java (hunk 1 above)
Comment (Spaghetti-12) [reply src/main/java/BuyStockUseCase/BuyInputRequest.java]: idk how I missed the other 2, fixed it now

File: src/main/java/APIInterface/StockAPIAccess.java [hunk 3]
```diff
@@ -4,7 +4,7 @@
 import yahoofinance.YahooFinance;
 import java.io.IOException;
 
-public class StockAPIAccess {
+public class StockAPIAccess implements StockDatabaseAccess{
     public StockAPIResponse accessAPI(StockAPIRequest req) throws IOException {
```
Comment (eladperlman) [inline src/main/java/APIInterface/StockAPIAccess.java:8]: just add the @override tag

Comment (Spaghetti-12) [reply src/main/java/APIInterface/StockAPIAccess.java:9]: fixed

**Review by eladperlman — APPROVED** [review]
Approving

**Comment by noel-toms** [comment]
As Elliot mentioned we should watching for exceptions whenever we make a call to the API. For example, in the StockAPIRequest class the getSymbol method should have a catch for invalid symbols. That being said I think an important note to consider is having custom exceptions where we know why it was thrown. In the given example we know that the error is because of an invalid  symbol. Custom exceptions will help us better identify problems when interacting with the APIInterface.

**Review by noel-toms — APPROVED** [review]

**Review by benmarlow — APPROVED** [review]
It all looks good to me. Approved

**Pull request merged**